src/PostOffice.java

```diff
@@ -0,0 +1,44 @@
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Objects;
+
+public class PostOffice {
+
+    private ArrayList<Mail> packages;
```

Comment:
I suggest using List , which is more general and implements the Abstraction OOP principle better. So you don't force this specific implementation on the class